core/win32/inject.c

```diff
@@ -1083,20 +1082,25 @@ generate_switch_mode_jmp_to_hook(HANDLE phandle, byte *local_code_buf,
     instr_t *restore_esp =
         INSTR_CREATE_mov_ld(GDC, opnd_create_reg(REG_ESP),
                             OPND_CREATE_MEM32(REG_NULL, (int)(size_t)mode_switch_data));
-
+	byte* eaxpos =(byte*) ((size_t)mode_switch_data) + 8;
+	// Restoring eax back, which is storing routine address that needs to be passed to RtlUserStartThread
```

Comment:
Where is eax clobbered, such that we need to restore it?  Not sure I understand this.

Reply:
I have no idea how/why/where EAX is getting clobbered in the process yet, but it is.

Reply:
Best to document that fact in a comment so future readers know what the status is.